xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/security/XarSecurityRule.java

```diff
@@ -78,4 +80,33 @@ public RuleState getState()
     {
         return RuleState.DENY;
     }
+
+    @Override
+    public int hashCode()
+    {
+        return new HashCodeBuilder(3, 17).append(right).append(simple)
```

Comment:
Why you do pass something to the HashCodeBuilder constructor ?

Reply:
I guess it's because it's the way documented in https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HEquals2FHashCodeandToStringimplementations ? :)

Reply:
OK, looks like a mistake to me since there is no point in doing that most of the time.

Reply:
Fixed in  d712af2